athena-neptune/src/test/java/com/amazonaws/athena/connectors/neptune/NeptuneMetadataHandlerTest.java

```diff
@@ -0,0 +1,251 @@
+import com.amazonaws.athena.connector.lambda.metadata.MetadataResponse;
+import com.amazonaws.athena.connector.lambda.security.FederatedIdentity;
+import com.amazonaws.athena.connector.lambda.security.LocalKeyFactory;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.Column;
+import com.amazonaws.services.glue.model.GetTableResult;
+import com.amazonaws.services.glue.model.GetTablesRequest;
+import com.amazonaws.services.glue.model.GetTablesResult;
+import com.amazonaws.services.glue.model.StorageDescriptor;
+import com.amazonaws.services.glue.model.Table;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.google.common.collect.ImmutableList;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.commons.collections.CursorableLinkedList.Cursor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NeptuneMetadataHandlerTest extends TestBase {
+    private static final Logger logger = LoggerFactory.getLogger(NeptuneMetadataHandlerTest.class);
+
+    @Mock
+    private AWSGlue glue;
+
+    @Mock
+    private GetTablesRequest glueReq = null;
+
+    private NeptuneMetadataHandler handler = null;
+
+    private boolean enableTests = System.getenv("publishing") != null
+            && System.getenv("publishing").equalsIgnoreCase("true");
+
+    private BlockAllocatorImpl allocator;
+
+    @Before
+    public void setUp() throws Exception {
+        // glue = mock(AWSGlue.class);
+
+        // glueReq = mock(GetTablesRequest.class);
+
+        // glueReq.setDatabaseName("TestDB");
+
+        logger.info("setUpBefore - enter");
+        allocator = new BlockAllocatorImpl();
+        handler = new NeptuneMetadataHandler(glue,
+                new NeptuneConnection(System.getenv("neptune_endpoint"), System.getenv("neptune_port")),
+                new LocalKeyFactory(), mock(AWSSecretsManager.class), mock(AmazonAthena.class), "spill-bucket",
+                "spill-prefix");
+        logger.info("setUpBefore - exit");
+    }
+
+    @After
+    public void after() {
+        allocator.close();
+    }
+
+    @Test
+    public void doListSchemaNames() {
+        if (!enableTests) {
```

Comment:
remove referrences to enableTests, that is just for the tutorial. I'll hold off on reviewing the tests until you've updated/addressed the other items in the non-test code.

Reply:
ok, will remove this

Reply:
done